### PR TITLE
.github/workflows: add workflow to update go.mod Nix SRI hash

### PR DESCRIPTION
**Pull request by danderson**
So that I just get a quick PR to approve and merge instead of periodically discovering that the SRI hash has bitrotted.

Signed-off-by: David Anderson <danderson@tailscale.com>

---

I borrowed the credentials of the license updater, hoping it'll work out.

### CODE REVIEW

File: .github/workflows/update-flakes.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+on:
+  # run action when a change lands in the main branch which updates go.mod. Also
+  # allow manual triggering.
+  push:
+    branches:
+      - main
+    paths:
+      - go.mod
+      - .github/workflows/go-licenses.yml
```
Comment (bradfitz) [inline .github/workflows/update-flakes.yml:11]: Stale

Comment (danderson) [reply .github/workflows/update-flakes.yml]: Fixed.

File: .github/workflows/update-flakes.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: go-licenses
```
Comment (bradfitz) [inline .github/workflows/update-flakes.yml:1]: go-update-flakes?

File: .github/workflows/update-flakes.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Get access token
+        uses: tibdex/github-app-token@f717b5ecd4534d3c4df4ce9b5c1c2214f0f7cd06 # v1.6.0
+        id: generate-token
+        with:
+          app_id: ${{ secrets.LICENSING_APP_ID }}
+          installation_id: ${{ secrets.LICENSING_APP_INSTALLATION_ID }}
+          private_key: ${{ secrets.LICENSING_APP_PRIVATE_KEY }}
+
+      - name: Send pull request
```
Comment (bradfitz) [inline .github/workflows/update-flakes.yml:37]: so this just does nothing if there are no changes?

**Review by bradfitz — APPROVED** [review]

**Review by maisem — APPROVED** [review]

File: .github/workflows/update-flakes.yml (hunk 2 above)
Comment (danderson) [reply .github/workflows/update-flakes.yml:1]: Bah, I'm bad at this copying and pasting. Fixed.

File: .github/workflows/update-flakes.yml (hunk 3 above)
Comment (danderson) [reply .github/workflows/update-flakes.yml:37]: I believe so, based on what the license updater workflow looks like.

**Pull request merged**